### ISSUE TITLE
Review deprecations for 0.56

### ISSUE THREAD

**Issue by stuartarchibald** [numba/numba#7770]
As title. Current known changed needed include:

 * CUDA: Deprecate support for CC < 5.3 and CTK < 10.2, xref https://github.com/numba/numba/pull/7755

**Comment by esc** [comment]
> * [ ]  Remove the `add_user_function()` function: https://numba.readthedocs.io/en/latest/reference/deprecation.html#deprecation-of-numba-core-base-basecontext-add-user-function

Looking at the `latest` docs at: https://numba.readthedocs.io/en/latest/reference/deprecation.html#deprecation-and-removal-of-numba-core-base-basecontext-add-user-function


<img width="1130" alt="Screen Shot 2022-06-23 at 14 11 27" src="https://user-images.githubusercontent.com/41563/175295939-536c4ba3-b8bb-49cb-80a8-18f002dcdc17.png">

The removal with 0.56 is already documented.

**Comment by esc** [comment]
> * [ ]  CUDA: Deprecate support for CC < 5.3 and CTK < 10.2, xref [CUDA: Deprecate support for CC < 5.3 and CTK < 10.2 #7755](https://github.com/numba/numba/pull/7755)

Looking at:

https://numba.readthedocs.io/en/latest/reference/deprecation.html#deprecation-and-removal-of-cuda-toolkits-10-2-and-devices-with-cc-5-3

<img width="875" alt="Screen Shot 2022-06-23 at 14 22 09" src="https://user-images.githubusercontent.com/41563/175297507-09869625-7b17-4d00-a8b8-86c154059496.png">

This has been taken care of already, it seems.

**Comment by gmarkall** [comment]
> > * [ ]   Eager compilation of device functions: https://numba.readthedocs.io/en/latest/reference/deprecation.html#deprecation-of-eager-compilation-of-cuda-device-functions
> 
> @gmarkall question: does the above need an entry for what happens with 0.56?

No, nothing's changed here, but I think it's worth leaving in for reference for now, because the change doesn't have an obvious breaking effect like "suddenly existing code doesn't compile" but it does have a side effect in terms of when things are compiled, so I think it's useful to have to refer to.

**Comment by gmarkall** [comment]
Note that #8191 also fixes up a couple of deprecation-related items for the `ptx` property and `inspect_ptx()`.

**Comment by gmarkall** [comment]
(BTW I looked into all the other things on this issue and I think they're all as they should be already, so I'm going to tick the boxes).
